### ISSUE TITLE
too_few_lines

### ISSUE THREAD

**Issue by JonathanWoollett-Light** [rust-lang/rust-clippy#11334]
### What it does

Checks for functions with a small amount of lines.

The counter-part to [too_many_lines](https://rust-lang.github.io/rust-clippy/master/index.html#/too_many_lines).

It should have a configuration variable `too-few-lines-threshold: u64` which should default to `1`.

It should be a pedantic lint (presuming a default threshold of `1`).

### Advantage

- Very small functions may increase lines of code by their usage. Their signature and calls may outweigh the code duplication they remove.

### Drawbacks

- Set at anything above very small (e.g 1) it can produce many unreasonable warnings.

### Example

```rust
fn main() {
    let a = add(1, 2);
}
fn add(rhs: u8, lhs: u8) -> u8 {
    rhs + lhs
}
```

Could be written as:

```rust
fn main() {
    let a = 1 + 2;
}
```

---

```rust
fn main() {
    let a = line(4, 3, 2, 1);
}
fn line(x: u8, a: u8, b: u8, c: u8) -> u8 {
    add(add(x.pow(a), mul(x, b)), c)
}
fn mul(rhs: u8, lhs: u8) -> u8 {
    rhs * lhs
}
fn add(rhs: u8, lhs: u8) -> u8 {
    rhs + lhs
}
```

Could be written as:

```rust
fn main() {
    let a = 4u8.pow(3) + (4*2) + 1;
}
```

**Comment by asquared31415** [comment]
Unfortunately this would need to be `restriction`, even `pedantic` would lint on way too much.  For example, a struct with private fields that have get-only accessors, implementations of most of the math operators on wrapper types, and so on.

**Comment by JonathanWoollett-Light** [comment]
Considering getters and setters I agree it makes sense for this to be restriction.

---

I would generally advocate against getters/setters and instead using a wrapper that only implements `Deref`and not `DerefMut` e.g.
```rust
struct MyStruct {
    a: u32,
    b: Immutable<u32>
}
```
